Incorrect type annotation on Pack.margin

### Describe the bug

The [documentation](https://toga.readthedocs.io/en/v0.5.0/reference/style/pack.html#margin) states that `margin` accepts ``<integer> or <tuple> of length 1-4``. However the type annotation only allows `int` and a `tuple` of a single `int`.

### Steps to reproduce

1\. Attempt to use `Pack` like in the [tutorial](https://docs.beeware.org/en/latest/tutorial/tutorial-2.html#adding-some-content-of-our-own). 
```python
name_label = toga.Label("Your name: ", style=Pack(margin=(0, 5)))
```
2\. Run `mypy` on the file.
3\. Get error: 
``` 
error: Argument "margin" to "Pack" has incompatible type "tuple[int, int]"; expected "int | tuple[int]"  [arg-type]
```


### Expected behavior

No `mypy` error.

### Screenshots

_No response_

### Environment

- Operating System: Ubuntu 24.04
- Python version: 3.12
- Software versions:
  - Briefcase: 0.3.22
  - Toga: 0.5.0


### Logs




### Additional context

_No response_

--- a/changes/3396.bugfix.rst
+++ b/changes/3396.bugfix.rst
@@ -1,1 +1,1 @@
-Fixed type annotation on `Pack.margin` to match documented types.
+The type annotation for directional style properties (``margin``, the deprecated alias ``padding``) has been corrected.

--- a/changes/3396.bugfix.rst
+++ b/changes/3396.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed type annotation on `Pack.margin` to match documented types.

--- a/core/src/toga/style/pack.py
+++ b/core/src/toga/style/pack.py
@@ -206,7 +206,13 @@ class Pack(BaseStyle):
     height: str | int = validated_property(NONE, integer=True, initial=NONE)
     flex: float = validated_property(number=True, initial=0)
 
-    margin: int | tuple[int] = directional_property("margin{}")
+    margin: (
+        int
+        | tuple[int]
+        | tuple[int, int]
+        | tuple[int, int, int]
+        | tuple[int, int, int, int]
+    ) = directional_property("margin{}")
     margin_top: int = validated_property(integer=True, initial=0)
     margin_right: int = validated_property(integer=True, initial=0)
     margin_bottom: int = validated_property(integer=True, initial=0)
@@ -249,7 +255,13 @@ class Pack(BaseStyle):
     # 2024-12: Backwards compatibility for Toga < 0.5.0
     ######################################################################
 
-    padding: int | tuple[int] = aliased_property(source="margin", deprecated=True)
+    padding: (
+        int
+        | tuple[int]
+        | tuple[int, int]
+        | tuple[int, int, int]
+        | tuple[int, int, int, int]
+    ) = aliased_property(source="margin", deprecated=True)
     padding_top: int = aliased_property(source="margin_top", deprecated=True)
     padding_right: int = aliased_property(source="margin_right", deprecated=True)
     padding_bottom: int = aliased_property(source="margin_bottom", deprecated=True)

